nncf/experimental/torch/sparsity/movement/layers.py

```diff
@@ -0,0 +1,292 @@
+
+
+@COMPRESSION_MODULES.register()
+class MovementSparsifier(nn.Module):
+    def __init__(
+        self,
+        target_module_node: NNCFNode,
+        sparse_cfg: SparseConfig = SparseConfig(mode=SparseStructure.FINE),
+        frozen: bool = True,
+        compression_lr_multiplier: Optional[float] = None,
+        layerwise_loss_lambda: float = 0.5,
+    ):
+        super().__init__()
+        self.target_module_node = target_module_node
+        self.prune_bias = (target_module_node.layer_attributes.bias not in (False, None))
+        self.frozen = frozen
+        self.layerwise_loss_lambda = layerwise_loss_lambda
+        self._importance_threshold = -math.inf
+        self._importance_regularization_factor = 0.
+
+        weight_shape = target_module_node.layer_attributes.get_weight_shape()
+        self.weight_ctx = BinaryMask(weight_shape)
+        self.sparse_factors = self._get_sparse_factors(weight_shape, sparse_cfg)
+        self.sparse_structure = sparse_cfg.mode
+
+        weight_importance_shape = self._get_weight_importance_shape(
+            weight_shape, self.sparse_factors, self.sparse_structure)
+        self.weight_importance = CompressionParameter(
+            torch.zeros(weight_importance_shape),
+            requires_grad=not self.frozen,
+            compression_lr_multiplier=compression_lr_multiplier,
+        )
+        self.weight_ctx.binary_mask = self._calc_training_binary_mask()
+
+        if self.prune_bias:
+            bias_shape = target_module_node.layer_attributes.get_bias_shape()
+            self.bias_ctx = BinaryMask(bias_shape)
+            bias_importance_shape = weight_importance_shape[0]
+            self.bias_importance = CompressionParameter(
+                torch.zeros(bias_importance_shape),
+                requires_grad=not self.frozen,
+                compression_lr_multiplier=compression_lr_multiplier,
+            )
+            self.bias_ctx.binary_mask = self._calc_training_binary_mask(is_bias=True)
+
+        self.mask_calculation_hook = MaskCalculationHook(self)
+
+    @property
+    def importance_threshold(self):
+        return self._importance_threshold
+
+    @importance_threshold.setter
+    def importance_threshold(self, value: float):
+        self._importance_threshold = value
+
+    @property
+    def importance_regularization_factor(self):
+        return self._importance_regularization_factor
+
+    @importance_regularization_factor.setter
+    def importance_regularization_factor(self, value: float):
+        self._importance_regularization_factor = value
+
+    def forward(self, weight: torch.Tensor, bias: Optional[torch.Tensor] = None
+                ) -> Tuple[torch.Tensor, Optional[torch.Tensor]]:
+        if is_tracing_state():
+            with no_jit_trace():
+                masked_weight = weight.mul(self.weight_ctx.binary_mask)
+                masked_bias = None if bias is None else bias.mul(self.bias_ctx.binary_mask)
```

Comment:
@ljaljushkin should we remove `no_jit_trace` as it was suggested in #1405?

Reply:
Yes, it should be removed after merge #1405

Reply:
@vuiseng9 @yujiepan-work https://github.com/openvinotoolkit/nncf/pull/1405 has been merged

Reply:
Please remove `no_jit_trace` for the movement sparsity